diff --git a/node_modules/react-native-wechat-lib/src/index.js b/node_modules/react-native-wechat-lib/src/index.js
index 3bb59a6..6ddb35e 100644
--- a/node_modules/react-native-wechat-lib/src/index.js
+++ b/node_modules/react-native-wechat-lib/src/index.js
@@ -4,11 +4,14 @@ import { DeviceEventEmitter, NativeModules, Platform } from 'react-native';
 import { EventEmitter } from 'events';

 let isAppRegistered = false;
-let { WeChat, WechatLib } = NativeModules;
+let { WeChat, WechatLib, RCTWeChat } = NativeModules;

 if (WeChat == null) {
   WeChat = WechatLib;
 }
+if (WeChat == null) {
+  WeChat = RCTWeChat;
+}

 // Event emitter to dispatch request and response from WeChat.
 const emitter = new EventEmitter();
