diff --git a/node_modules/react-native/Libraries/Components/Keyboard/KeyboardAvoidingView.js b/node_modules/react-native/Libraries/Components/Keyboard/KeyboardAvoidingView.js
index d0fb5c4..80b7b8b 100644
--- a/node_modules/react-native/Libraries/Components/Keyboard/KeyboardAvoidingView.js
+++ b/node_modules/react-native/Libraries/Components/Keyboard/KeyboardAvoidingView.js
@@ -155,7 +155,6 @@ class KeyboardAvoidingView extends React.Component<
       this._setBottom(0);
       return;
     }
-
     const {duration, easing, endCoordinates} = this._keyboardEvent;
     const height = await this._relativeKeyboardHeight(endCoordinates);
 
@@ -163,7 +162,7 @@ class KeyboardAvoidingView extends React.Component<
       return;
     }
 
-    this._setBottom(height);
+    this._setBottom(height < 50 ? 0 : height);
 
     const enabled = this.props.enabled ?? true;
     if (enabled && duration && easing) {
