diff --git a/node_modules/xcode/lib/pbxFile.js b/node_modules/xcode/lib/pbxFile.js
index 9553bfb..84d3854 100644
--- a/node_modules/xcode/lib/pbxFile.js
+++ b/node_modules/xcode/lib/pbxFile.js
@@ -47,7 +47,8 @@ var FILETYPE_BY_EXTENSION = {
         xcodeproj: 'wrapper.pb-project',
         xctest: 'wrapper.cfbundle',
         xib: 'file.xib',
-        strings: 'text.plist.strings'
+        strings: 'text.plist.strings',
+        storyboard: 'file.storyboard'
     },
     GROUP_BY_FILETYPE = {
         'archive.ar': 'Frameworks',
@@ -78,7 +79,8 @@ var FILETYPE_BY_EXTENSION = {
         'text.plist.xml': 4,
         'text.script.sh': 4,
         'text.xcconfig': 4,
-        'text.plist.strings': 4
+        'text.plist.strings': 4,
+        'file.storyboard': 4
     };
 
 
diff --git a/node_modules/xcode/lib/pbxProject.js b/node_modules/xcode/lib/pbxProject.js
index 068548a..9859a34 100644
--- a/node_modules/xcode/lib/pbxProject.js
+++ b/node_modules/xcode/lib/pbxProject.js
@@ -1679,7 +1679,7 @@ function correctForFrameworksPath(file, project) {
 function correctForPath(file, project, group) {
     var r_group_dir = new RegExp('^' + group + '[\\\\/]');
 
-    if (project.pbxGroupByName(group).path)
+    if (project.pbxGroupByName(group)?.path)
         file.path = file.path.replace(r_group_dir, '');
 
     return file;
@@ -1929,7 +1929,7 @@ pbxProject.prototype.getPBXGroupByKey = function(key) {
 };
 
 pbxProject.prototype.getPBXVariantGroupByKey = function(key) {
-    return this.hash.project.objects['PBXVariantGroup'][key];
+    return this.hash.project.objects['PBXVariantGroup']?.[key];
 };
 
 
