diff --git a/node_modules/@babel/helper-define-polyfill-provider/lib/visitors/usage.js b/node_modules/@babel/helper-define-polyfill-provider/lib/visitors/usage.js
index 77def9f..1ff461f 100644
--- a/node_modules/@babel/helper-define-polyfill-provider/lib/visitors/usage.js
+++ b/node_modules/@babel/helper-define-polyfill-provider/lib/visitors/usage.js
@@ -10,7 +10,7 @@ function isRemoved(path) {
     var _path$parentPath$node;
     if (!((_path$parentPath$node = path.parentPath.node) != null && (_path$parentPath$node = _path$parentPath$node[path.listKey]) != null && _path$parentPath$node.includes(path.node))) return true;
   } else {
-    if (path.parentPath.node[path.key] !== path.node) return true;
+    if (path.parentPath.node?.[path.key] !== path.node) return true;
   }
   return isRemoved(path.parentPath);
 }
